RecoLocalTracker/SiStripClusterizer/plugins/SiStripApprox2Clusters.cc

```diff
@@ -23,23 +26,38 @@ class SiStripApprox2Clusters : public edm::global::EDProducer<> {
 
 private:
   edm::EDGetTokenT<edmNew::DetSetVector<SiStripApproximateCluster>> clusterToken_;
+  edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> tkGeomToken_;
 };
 
 SiStripApprox2Clusters::SiStripApprox2Clusters(const edm::ParameterSet& conf) {
   clusterToken_ = consumes<edmNew::DetSetVector<SiStripApproximateCluster>>(
       conf.getParameter<edm::InputTag>("inputApproxClusters"));
+  tkGeomToken_ = esConsumes();
   produces<edmNew::DetSetVector<SiStripCluster>>();
 }
 
 void SiStripApprox2Clusters::produce(edm::StreamID id, edm::Event& event, const edm::EventSetup& iSetup) const {
   auto result = std::make_unique<edmNew::DetSetVector<SiStripCluster>>();
   const auto& clusterCollection = event.get(clusterToken_);
 
+  const auto& tkGeom = &iSetup.getData(tkGeomToken_);
+  const auto tkDets = tkGeom->dets();
```

Comment:
Why do you define `tkDets` here if you never use it? The one you use in `std::find_if` is defined in [line52](https://github.com/cms-sw/cmssw/pull/38735/files#diff-436257e253b4870ab2d4471abac3b954a10bd8c05196f7200bd5e2b50bd03f78R52), right?

Reply:
remove the one in the loop, not needed.